nni/retiarii/experiment/pytorch.py

```diff
@@ -76,8 +77,9 @@ def _validation_rules(self):
 
 
 class RetiariiExperiment(Experiment):
-    def __init__(self, base_model: Model, trainer: BaseTrainer,
-                 applied_mutators: Mutator = None, strategy: BaseStrategy = None):
+    def __init__(self, base_model: nn.Module, trainer: Union[TrainingConfig, BaseOneShotTrainer],
+                 applied_mutators: List[Mutator] = None, strategy: BaseStrategy = None):
+        # TODO: The current design of init interface of Retiarii experiment needs to be reviewed.
```

Comment:
`trainer` could be both configuration and instantiated trainer?

Reply:
Have to admit that this is very confusing. Maybe we can find a better name than "TrainingConfig".